### PR TITLE
Consolidate `TcpOpsIncGen` to include attrdef and enum decls/defs

### PR DESCRIPTION
**Pull request by sjain-stanford**
This is idiomatic to usage in upstream MLIR. Includes other minor cleanup.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -91,28 +47,34 @@ gentbl_cc_library(
...
-    srcs = [
-        "lib/Dialect/IR/TcpDialect.cpp",
-        "lib/Dialect/IR/TcpOps.cpp",
-    ],
-    hdrs = [
-        "include/mlir-tcp/Dialect/IR/TcpDialect.h",
-        "include/mlir-tcp/Dialect/IR/TcpOps.h",
-    ],
+    srcs = glob(["lib/Dialect/IR/*.cpp"]),
+    hdrs = glob(["include/mlir-tcp/Dialect/IR/*.h"]),
```
Comment (sanjoy) [inline BUILD:75]: Nit: can we avoid `glob`s here?  They tend to accidentally pull in unnecessary files.

**Review by sanjoy — APPROVED** [review]
Minor nit inline.

File: BUILD (hunk 1 above)
Comment (sjain-stanford) [reply BUILD]: Sure, fIxed this and all other uses of `glob` in main BUILD.

File: lib/Conversion/TorchToTcp/TorchToTcp.cpp [hunk 2]
```diff
@@ -75,8 +75,7 @@ class ConvertTorchToTcp : public ConvertTorchToTcpBase<ConvertTorchToTcp> {
 
 } // namespace
 
-std::unique_ptr<OperationPass<func::FuncOp>>
-mlir::tcp::createConvertTorchToTcpPass() {
+std::unique_ptr<OperationPass<func::FuncOp>> createConvertTorchToTcpPass() {
```
Comment (sjain-stanford) [inline lib/Conversion/TorchToTcp/TorchToTcp.cpp:78]: Fix for:
```
INFO: From Compiling lib/Conversion/TorchToTcp/TorchToTcp.cpp [for host]:                                                                                                                       
lib/Conversion/TorchToTcp/TorchToTcp.cpp:79:12: warning: extra qualification on member 'createConvertTorchToTcpPass' [-Wextra-qualification]                                                    
mlir::tcp::createConvertTorchToTcpPass() {                                                                                                                                                      
           ^                                 
```

**Pull request merged**